erdos/src/configuration.rs

```diff
@@ -28,11 +29,11 @@ pub struct Configuration {
 }
 
 impl Configuration {
-    /// Creates a new node configuration.
+    /// Creates a new [`Worker`](crate::node::WorkerHandle) configuration.
     pub fn new(
-        node_index: NodeId,
-        data_addresses: Vec<SocketAddr>,
-        control_addresses: Vec<SocketAddr>,
+        id: WorkerId,
+        leader_address: SocketAddr,
+        data_plane_address: SocketAddr,
```

Comment:
Does the worker's data plane address need to be specified? IMO this can be assigned by the OS automatically and registered with the leader.

For example, [passing port 0 binds tokio's `TcpListener` to an OS-assigned port](https://docs.rs/tokio/latest/tokio/net/struct.TcpListener.html).

Reply:
The `Configuration` struct seems to be constructed mostly using its `from_args` method, and [we default to a randomly chosen address in `src/lib.rs`](https://github.com/sukritkalra/erdos/blob/data-plane/erdos/src/lib.rs#L226). The option was left there so that a user could choose to assign a port if they so wish.

Reply:
Awesome. Can we document this behavior in the command line args and the configuration constructor?